src/extract.c

```diff
@@ -188,19 +191,19 @@ bool
 detect_sparse_region(const void *data, size_t size, size_t *len_ret)
 {
 	const void *p = data;
-	const void * const end = data + size;
+	const void * const end = POINTER_FIX()data + size;
 	size_t len = 0;
 	bool zeroes = false;
 
 	while (p != end) {
-		size_t n = min(end - p, SPARSE_UNIT);
+		size_t n = min(POINTER_FIX()end - POINTER_FIX()p, SPARSE_UNIT);
```

Comment:
I went for a slightly different macro for the same issue, where I used:
```C
#ifdef _MSC_VER
#define _PTR(x)	(void*)((uintptr_t)x)
#else
#define _PTR(x)	x
#endif
```
For pointer arithmetic, and then could use a less conspicuous:
```C
    size_t n = min(_PTR(end - p), SPARSE_UNIT);
```